#! /usr/bin/env python
# Task Long configuration
import math

PACKAGE='task_manager_uavsim'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *
from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("goal_x",        double_t, 0,    "X coordinate of destination",  0.)
gen.add("goal_y",        double_t, 0,    "Y coordinate of destination",  0.)
gen.add("goal_z",        double_t, 0,    "Z coordinate of destination",  1., 0.0, 100.0)
gen.add("goal_heading",  double_t, 0,    "Heading coordinate at destination",  0., -math.pi, math.pi)
gen.add("max_velocity",  double_t, 0,    "Max allowed velocity",  1.0)

exit(gen.generate(PACKAGE, "task_manager_uavsim", "TaskGoTo"))

